06_neighborhoods/Transportation/transit_trips_cost_city.R

```diff
@@ -0,0 +1,283 @@
+transit_trips_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, transit_trips_80ami)
+transit_cost_tracts_2019 <- transport_tracts_2019 %>%
+  select(GEOID, state, county, tract, blkgrps, population, households, t_80ami)
+
+
+###################################################################
+
+# 2. Crosswalk to the cities (from tracts) & aggregate data
+
+# bring in the crosswalk of interest (we want the year range to be before 2020, when tract definitions changed)
+tract_place <- read_csv("geographic-crosswalks/data/geocorr2018_tract_to_place.csv")
+# remove the decimal from the tract variable
+tract_place <- tract_place %>%
+  mutate(tract = tract*100)
+# add in leading zeroes as needed
+tract_place <- tract_place %>%
+  mutate(state = sprintf("%0.2d", as.numeric(state)),
+         tract = sprintf("%0.6d", as.integer(as.character(tract))),
+         placefp = sprintf("%0.5d", as.numeric(placefp)),
+         county = substr(county, nchar(county) - 2, nchar(county))
+  ) %>%
+  rename(place = placefp)
+
+
+# NOW, aggregate the tract-level data to the place levels for each variable for each year
+
+# For when we want to: Limit to the Census Places we want 
+# first, bring in the places crosswalk (place-populations.csv)
+places <- read_csv("geographic-crosswalks/data/place-populations.csv")
+# keep only the most relevant year
+places <- places %>%
+  filter(year == 2019)
+
+
+# TRANSPORTATION COST 2015
+transit_cost_city_2015 <- left_join(transit_cost_tracts_2015, tract_place, by=c("state", "tract"))
+# collapse to places and also create data quality marker
+# data quality can be 1 when most of the tracts that fall in the place (e.g., >50% of the tracts) 
+# have most of their area falling in the place (e.g., >50% of the tract's area is in the place)
+# otherwise, data quality is 2
+transit_cost_city_2015 <- transit_cost_city_2015 %>% 
+  dplyr::group_by(state, place) %>% 
+  dplyr::summarize(index_transportation_cost = round(mean(t_80ami), 2),
+                   n = n(),
+                   dq = sum(afact > 0.5, na.rm = TRUE)
+  )%>%
+  dplyr::mutate(index_transportation_cost_quality = ifelse(dq / n > 0.5, 1, ifelse(dq / n <= 0.5, 2, NA)))
+
+
+# left join with places file to get rid of irrelevant places data
+transit_cost_city_2015 <- left_join(places, transit_cost_city_2015, by=c("state","place"))
+# 28313 obs to 486 obs
+
+# keep only the variables we will need & replace NA qual vars where there is NA metric value
+transit_cost_city_2015 <- transit_cost_city_2015 %>% 
+  mutate(year=2015)%>%
+  select(year, state, place, index_transportation_cost, index_transportation_cost_quality)%>%
+  dplyr::mutate(index_transportation_cost_quality = ifelse(is.na(index_transportation_cost), NA, index_transportation_cost_quality))
+
+
+
+# TRANSPORTATION COST 2019
+transit_cost_city_2019 <- left_join(transit_cost_tracts_2019, tract_place, by=c("state", "tract"))
```

Comment:
Same 🚨🚨🚨 issue as above. Instead of joining by two variables a better practice would be to merge based on a unique geo ID in the format of: ssccctttttt. 

I believe everything below that uses transit_cost_city_2019 is incorrect.

Reply:
Yes -- new df count is 123825 obs

Reply:
Merge looks correct now.